modules/apps/client-extension/client-extension-web/src/main/resources/META-INF/resources/js/components/FileEntryPicker.tsx

```diff
@@ -0,0 +1,351 @@
+	displayType: 'info',
+	message: '',
+};
+
+const FileEntryPicker = ({
+	companyId,
+	externalReferenceCode,
+	frontendTokenDefinition,
+	frontendTokenDefinitionFileName,
+	namespace,
+}) => {
+	const [feedback, setFeedback] = useState(EMPTY_FEEDBACK);
+	const [fileName, setFileName] = useState(frontendTokenDefinitionFileName);
+	const [alertRole, setAlertRole] = useState<'alert' | null>(null);
+	const [isValidatingJSON, setIsValidatingJSON] = useState(false);
+
+	const frontendTokenDefinitionRef = useRef<HTMLTextAreaElement>();
+	const inputFileRef = useRef<HTMLInputElement>();
+	const selectFileButtonRef = useRef<HTMLButtonElement>();
+
+	const setFrontendTokenDefinitionRefValue = (value: string) => {
+		if (frontendTokenDefinitionRef.current) {
+			frontendTokenDefinitionRef.current.value = value;
+		}
+	};
+
+	const clearSelection = () => {
+		setFileName('');
+
+		setFeedback(EMPTY_FEEDBACK);
+
+		setFrontendTokenDefinitionRefValue('');
+
+		selectFileButtonRef.current?.focus();
+	};
+
+	const getButtonSideLabel = (
+		fileName: string,
+		isValidatingJSON: boolean
+	) => {
+		if (isValidatingJSON) {
+			return Liferay.Language.get('validating-json');
+		}
+
+		if (fileName) {
+			return fileName;
+		}
+
+		return Liferay.Language.get('no-json-selected');
+	};
+
+	const onInputChange = async (event: any) => {
+		const target = event.target as HTMLInputElement;
+
+		const filePath = target.value;
+
+		const fileName = filePath.replace(/^.*[\\]/, '');
+
+		setFileName(fileName);
+
+		if (!fileName.endsWith('.json')) {
+			setFeedback({
+				displayType: 'danger',
+				message: Liferay.Language.get(
+					'the-format-is-not-valid-please-upload-a-valid-frontend-token-definition-json-file'
+				),
+			});
+
+			setFrontendTokenDefinitionRefValue(MALFORMED_JSON);
+
+			return;
+		}
+
+		setIsValidatingJSON(true);
+
+		if (!alertRole) {
+			setAlertRole('alert');
+		}
+
+		setFeedback({
+			displayType: 'info',
+			message: Liferay.Language.get(
+				'the-frontend-token-definition-json-file-is-being-uploaded-and-validated'
+			),
+		});
+
+		readInputFile(fileName, target);
+	};
+
+	function readInputFile(fileName: string, input: HTMLInputElement) {
+		if (input.files === null) {
+			return;
+		}
+
+		let file = input.files[0];
+
+		const fileReader = new FileReader();
+
+		fileReader.onload = async function (event) {
+			let frontendTokenDefinitionString = event.target?.result as string;
+
+			if (frontendTokenDefinitionString === '') {
+				file = new File(['{}'], fileName, {
+					type: 'application/json',
+				});
+				frontendTokenDefinitionString = '{}';
+			}
+
+			if (frontendTokenDefinitionRef.current) {
+				frontendTokenDefinitionRef.current.value = frontendTokenDefinitionString;
+			}
+
+			await handleUpload(file);
+
+			setIsValidatingJSON(false);
+		};
+
+		fileReader.readAsText(file);
+	}
+
+	const handlePost = async (body: any, endpoint: string) => {
+		try {
+			const response = await Liferay.Util.fetch(
```

Comment:
We should be importing fetch from frontend-js-web instead of using the global instance.